src/dotnet/ErrorAnalyzers/Core/src/Azure.Tools.ErrorAnalyzers/RuleAnalyzer.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Tools.ErrorAnalyzers
+{
+    /// <summary>
+    /// High-performance rule analyzer that maps error codes to fixes.
+    /// Immutable and thread-safe by design.
+    /// </summary>
+    public sealed class RuleAnalyzer
+    {
+        public RuleAnalyzer(string ruleId)
+        {
+            ArgumentException.ThrowIfNullOrWhiteSpace(ruleId);
+            RuleId = ruleId;
+        }
+
+        public string RuleId { get; }
+
+        public bool CanFix(RuleError error)
+        {
+            ArgumentNullException.ThrowIfNull(error);
+            return string.Equals(error.type, RuleId, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public Fix? GetFix(RuleError error)
+        {
+            if (!CanFix(error))
+                return null;
+
+            return AnalyzerPromptProvider.TryGetPrompt(RuleId, out var prompt) &&
```

Comment:
It feels like these should be local to each other and not two separate calls.  We should talk over the design here.